### PR TITLE
Ammolathe [WIP]

### PR DESCRIPTION
**Pull request by jakmak6**
Due to the weirdness of getting ammo and the lack of any kind of real ammunition diversity i have coded the Ammolathe. The ammolathe basically being like the way the protolathe and circuit imprinter are in that they are accessed via console and can be synced with R&D server to produce ammunition.

Currently the ammolathe functions as intended and works basically like a reskinned protolathe (because it kinda is) with its own recipes

However due to the general awkwardness of the price of ammunition and general balance i have instead left placeholders and i instead ask the rest of you to help me brainstorm what can be made with what research levels and for what material price.

I do not intend for the ammolathe to be able to make guns since that would be OP. This machine is meant to make ammo, grenades, and weapons chargers and firearm accessories.

### CODE REVIEW

File: code/modules/research/designs/ammolathe/miscammo.dm [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	name = "recharger kit"
+	desc = "A kit to building a weapons charger"
+	id = "recharger kit"
+	req_tech = list(Tc_COMBAT = 3, Tc_MATERIALS = 5) // suggestions for tech levels n shit
+	build_type = AMLATHE
+	materials = list(MAT_IRON = 5000, MAT_GLASS = 10000) // suggestions for material price
+	category = "miscammo"
+	build_path = /obj/item/device/recharger_kit // the entire original point of this PR was to make these things
+
+//	i swear on my life that in the final revision of the PR i will remove the shitposting
```
Comment (jknpj) [inline code/modules/research/designs/ammolathe/miscammo.dm:21]: Why

**Comment by strawberrytan** [comment]
Will the ammo printing capability of the protolathe be moved over to this?

File: code/modules/research/designs.dm [hunk 2]
```diff
@@ -52,6 +52,7 @@ The required techs are the following:
 #define PODFAB		32  //Used by the spacepod part fabricator. Same idea as the mechfab
 #define FLATPACKER	64  //This design creates a machine, not an item.
 #define GENFAB		128 //Generic item.
+#define AMLATHE		256 //Used by the ammolathe. makes ammo and grenades
```
Comment (SonixApache) [inline code/modules/research/designs.dm:55]: why not `MOLATHE` or just a full `AMMOLATHE`

**Comment by SonixApache** [comment]
this is gross

where's the ammo scan option

File: code/modules/research/designs.dm (hunk 2 above)
Comment (jakmak6) [reply code/modules/research/designs.dm:55]: AMMOLATHE looked better and easier to identity

**Comment by jakmak6** [comment]
@strawberrytan yes it will & more
And i misread your comment like a idiot.
The protolathe will keep its simple ammo printing but this new machine will instead be able to make ammo in bulk, pre loaded magazines, obscure ammo types, ammo accessories, and unique R&D research level ammo (for the cost of minerals ofcourse)

File: code/modules/research/designs/ammolathe/miscammo.dm (hunk 1 above)
Comment (jakmak6) [reply code/modules/research/designs/ammolathe/miscammo.dm:21]: the idea from the PR originally came from https://github.com/vgstation-coders/vgstation13/issues/15104 & discord shitposting.

**Comment by Riekland** [comment]
needing an R&D console is gay and stupid
I hate those things
it would also mean making an ammo lathe for somewhere other than science ie the places that are supposed to have ammo like the brig would require access changing for people to use it

**Comment by FudgePucker** [comment]
Can't it just be like the exosuit fab and not require an R&D computer?

**Comment by jakmak6** [comment]
riekland i had trouble making it work as standalone machine but it got real weird to do. Also it would need a R&D console to sync with the R&D database.

That and i was planning on having it be a roundstart thing for sec located in the armory/back of brig with its on sec access only r&d console

**Comment by Aurx** [comment]
Agreed that this should be standalone.

**Comment by Riekland** [comment]
I guess
I don't know how spagetti the R&D shit code is but I can only assume it's fucking awful given how every single R&D/research machine operates on a wholly different design philosophy
If it's actually feasable it would be very nice to have them just like an autolathe
Maybe let them eat design disks with ammo research on them to add more recipies?

**Comment by BobdaBiscuit** [comment]
I'm having flashbacks to D20station.

**Comment by Shadowmech88** [comment]
No progress or even discussion has been made on this in nearly four months.
@jakmak6 let me know if or when you wish to resume working on this.

**Pull request closed without merging**